FEC-11914 Bookmarks are not sent when the ks is set on the media options alone.

x-NR-x opened the pull request:
Update to Phoenix Analytics was not performed when the ks was set in the media, therefore bookmarks where not being sent.

Added code to not override Phoenix Analytics config if it was set in the player options upon init.
Added check to not override the kava plugin config in the player options if we received them from the client.

Added KS update to the player options in KalturaPlayer.
Fix in loadPlaylist function to fetch for a first ks occurrence and not only to take the ks from the first media option.

Fixed logic to update Kava and Phoenix plugin for specific media then if any custom plugin was sent update the specific config and not to override all configs.

Need to send the current mediaOptions in playlist for the plugins.

Updated logic in Playlist to save the whole media entry retrieved and not just it's sources, the metaData is needed as well.

Solves FEC-11914
Solves FEC-11913
Solves FEC-10249

x-NR-x committed:
FEC-11614 Fix, update to Phoenix Analytics was not performed when the ks was set in the media, therefore bookmarks where not being sent.
Added code to not override Phoenix Analytics config if it was set in the player options upon init.

x-NR-x committed:
FEC-11614 Missing code to previous commit, the pluginConfig was not sent, therefore update to Phoenix Analytics was not called.

chausov commented on Sources/OTT/KalturaOTTPlayer.swift:
```diff
@@ -106,8 +108,10 @@ import PlayKitKava
         
-        if let config = pluginConfig?.config,
-           !config.keys.contains(PhoenixAnalyticsPlugin.pluginName) {
+        let playerKSNotSet = playerOptions.ks?.isEmpty ?? true
```
```suggestion
var playerKSAvailable = false
var mediaKSAvailable = false

if let playerKs = playerOptions.ks {
    playerKSAvailable = !playerKs.isEmpty
}

if let mediaKs = mediaOptions?.ks {
    mediaKSAvailable = !mediaKs.isEmpty
}

// In cases player KS is NOT available we still may update Phoenix Analytics if media KS is available.
let shouldUpdatePhoenixAnalytics = !playerKSAvailable && mediaKSAvailable
```

x-NR-x replied:
Was changed

x-NR-x committed:
Merge branch 'develop' into FEC-11914

* develop:
  FEC-11857 PhoenixMediaProvider networkProtocol changed to be required… (#45)

x-NR-x committed:
Added check to not override the kava plugin config in the player options if we received them from the client.

x-NR-x committed:
Reverted sending the plugin config.

x-NR-x committed:
FEC-10249 Added KS update to the player options in KalturaPlayer.
Fix in loadPlaylist function to fetch for a first ks occurrence and not only to take the ks from the first media option.

x-NR-x committed:
FEC-10249 Missing code from previous commit to update the player options ks upon new media options ks.

x-NR-x committed:
FEC-11914 Fixed logic to update Kava and Phoenix plugin for specific media then if any custom plugin was sent update the specific config and not to override all configs.

x-NR-x committed:
FEC-11914 We need to return the OTTMediaOptions with all data for the analytics.

x-NR-x committed:
FEC-11914 Need to send the current mediaOptions in playlist for the plugins.

x-NR-x committed:
FEC-11914 Updated logic in Playlist to save the whole media entry retrieved and not just it's sources, the metaData is needed as well.

x-NR-x committed:
FEC-10249 KS updates for OVP Player.
FEC-11913 Fix to update plugin configs and not to override existing ones.

chausov commented on Sources/Playlist/KPPlaylistController.swift:
```diff
@@ -372,13 +377,19 @@ import PlayKit
-                entry.sources = loadedEntry?.sources
+                // Update the entry we got back from the provider
+                if let entryIndex = self.entries.firstIndex(of: entry) {
+                    self.entries[entryIndex] = mediaEntry
```
What are the benefits of changing whole entry and not only sources?

x-NR-x replied:
We need other values as well and not just the sources.

x-NR-x committed:
Merge branch 'develop' into FEC-11914

* develop:
  FEC-11683 OTT - Setting the fix of the serverURL in the setter so that it will be fixed for all requests. (#50)

x-NR-x committed:
FEC-11913 Updated code to update the PKMediaEntry object and not to replace it with a new object.
